lib/travis/remote_vcs/user.rb

```diff
@@ -25,7 +25,7 @@ def authenticate(provider: :github, code:, redirect_uri:)
         end
         return JSON.parse(resp.body)['data'] if resp.success?
```

Comment:
Looks like the same pattern appears over and over, can it be extracted into some method?
```
def perform_request(method, *args)
  resp =  connection.public_send(method, *args)
  raise ResponseError.new("Unexpected response: #{resp.body} #{resp.status}")

  resp.body
end
```